### DISCUSSION TITLE
I am trying to overwrite default Alarm setting.yaml

### DISCUSSION THREAD

**Discussion by alifaraz7178** [apache/skywalking#12289] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

I am trying to override the default alarm-setting.yaml using helm chart, I can see the dynamic config map created as below and I can also see the same getting picked by the oap pods but still on UI i don't see the changes in alarming its still the default one like if you see I have added tag but the same does not appear on UI.

Config Map

data:
  alarm.default.alarm-settings: |
    rules:
      service_resp_time_rule:
        metrics-name: service_resp_time
        op: ">"
        threshold: 1000
        period: 10
        count: 3
        silence-period: 5
        tags:
          level: WARNING
        message: Response time of service {name} is more than 1000ms in 3 minutes of last 10 minutes.
      service_sla_rule:
        metrics-name: service_sla
        op: "<"
        threshold: 8000
        period: 10
        count: 2
        silence-period: 3
        tags:
          level: WARNING
        message: Successful rate of service {name} is lower than 80% in 2 minutes of last 10 minutes
        
        Is adding webhook setting is important to make this work if yes can you share the sample webhook details as helm chart does notexplain well how to add them.
        
        

### What you expected to happen

Not sure

### How to reproduce

I am using OAP 9.4.0 using helm chart

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by alifaraz7178** [comment]
![Screenshot 2024-06-03 at 9 22 02 AM](https://github.com/apache/skywalking/assets/58845616/228a5e89-2d41-49a8-995f-76a339f9feeb)

**Comment by alifaraz7178** [comment]
rg.apache.skywalking.oap.server.core.alarm.provider.AlarmRulesWatcher 104 [pool-10-thread-1] INFO  [] - Update alarm rules to Rules(rules=[AlarmRule(alarmRuleName=service_resp_time_rule, metricsName=service_resp_time, includeNames=[], includeNamesRegex=, excludeNames=[], excludeNamesRegex=, includeLabels=[], includeLabelsRegex=, excludeLabels=[], excludeLabelsRegex=, threshold=1000, op=>, period=10, count=3, silencePeriod=5, message=Response time of service {name} is more than 1000ms in 3 minutes of last 10 minutes., onlyAsCondition=false, tags={level=WARNING}), AlarmRule(alarmRuleName=service_sla_rule, metricsName=service_sla, includeNames=[], includeNamesRegex=, excludeNames=[], excludeNamesRegex=, includeLabels=[], includeLabelsRegex=, excludeLabels=[], excludeLabelsRegex=, threshold=8000, op=<, period=10, count=2, silencePeriod=3, message=Successful rate of service {name} is lower than 80% in 2 minutes of last 10 minutes, onlyAsCondition=false, tags={level=WARNING})

**Comment by wu-sheng** [comment]
>  Is adding webhook setting is important to make this work if yes can you share the sample webhook details as helm chart does notexplain well how to add them.

It isn't. Alarm core doesn't rely on any hook.

**Comment by alifaraz7178** [comment]
@wu-sheng thanks for your prompt response. But then why this is not getting reflected on UI I have also attached by pod logs, whenever I update the config map i can see an update in pod but the changes never happens means I can still see the default alarm-setting.yaml in place as well as all default alarms are in place.

**Reply by wu-sheng** [reply]
I can't tell just from here.

**Reply by alifaraz7178** [reply]
@wu-sheng Thanks for prompt response, but what is the expected behaviour that it will update the default alarm-setting.yaml. Do you have any reference of Config map or any implementation which I can refer because I read each and documents and non of them is working
